src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Setup/MissingComponentProvider.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Setup;
+
+/// <summary>
+/// Determines the VS setup component requirements of the configured project and provides them to <see cref="IMissingSetupComponentRegistrationService"/>/>.
+///     Tracks the set of missing .NET workloads for a configured project.
+///     
+/// Detect during solution load the Net Core runtime version based on the target framework 
+/// used in a project file and pass this version to a service that can install the
+/// runtime component if not installed.
+/// </summary>
+[Export(ExportContractNames.Scopes.ConfiguredProject, typeof(IProjectDynamicLoadComponent))]
+[AppliesTo(ProjectCapability.DotNet)]
+internal sealed class MissingComponentProvider : OnceInitializedOnceDisposedAsync, IProjectDynamicLoadComponent
```

Comment:
Maybe this is coming in a future commit, but since this won't dynamically load and unload (as the .NET capability is static) would it make sense to drop the implementation of `IProjectDynamicLoadComponent`? I'm concerned a future dev may not understand why this does work once that should (in general for an `IProjectDynamicLoadComponent`) be done every time the capability shows up.

On the other hand I'm not sure there's an alternative interface with the narrower behavior we need here.

Reply:
Originally this was an `IPackageService`, but those are initialised very late (sometimes minutes after the user starts coding) and we want to provide feedback about missing components as early as possible so that problems don't go unexplained.

Looking through our other dynamic load components, there are a few approaches we take. For example, `SDKVersionTelemetryServiceComponent` derives from `AbstractMultiLifetimeComponent`, though that seems excessive for a static capability. Further, that component is performing low priority work (sending telemetry about the SDK version) which should be postponed until later (possibly even later than package load).

As we look at solution load performance, it may be that improving the scheduling of these components is helpful.